src/app/product-list/product-list.component.html

```diff
@@ -0,0 +1,13 @@
+<app-product [p] = "products"></app-product>
```

Comment:
why are you sending the whole "products" list to app-product ? app-product should take care of single product and not a list of products.

Iterate through the products here and each product to <app-product> component.
Eg: 
<app-product *ngFor="let product of products" [product]="product"></app-product>